### PR TITLE
Introduce PayeezyGateway

### PR DESCRIPTION
**Pull request by huoxito**
Hi all,

I believe [this gateway](https://developer.payeezy.com/) is somehow related to the existing FirstdataE4 gateway here. I took / copied the test examples from that one to write the tests for this one.

For the record, the PayeezyGateway class here was originally created by Sachin Shetty. I only removed dependencies and added test examples. Please let me know if there's something wrong regarding active_merchant compatibility when you have a chance.

cheers

### CODE REVIEW

**Comment by JDutil** [comment]
@huoxito should probably add to the README as well.

**Comment by huoxito** [comment]
just amended the commit with README change and a couple other fixes after running this in production, pls let me know if I could help with anything else to get this merged

File: lib/active_merchant/billing/gateways/payeezy.rb [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+      def authorize(money, creditcard, options = {})
+        commit(:authorize, purchase_or_auth_request(money, creditcard, options))
+      end
+
+      def capture(money, authorization, options = {})
+        commit(:capture, capture_or_credit_request(money, authorization, options))
+      end
+
+      def credit(money, authorization, options = {})
```
Comment (ntalbott) [inline lib/active_merchant/billing/gateways/payeezy.rb]: This should be `refund` not `credit`. `credit` does an unreferenced push of funds to a card.

**Comment by huoxito** [comment]
thanks @ntalbott, I've rebased on top of master again and renamed the method properly

**Pull request closed without merging**